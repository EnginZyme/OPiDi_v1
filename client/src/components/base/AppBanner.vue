<template>
  <div class="app-banner__grid">
    <div :class="dark ? 'app-banner--dark' : 'app-banner--light'">
      <img
        v-if="showImage && dark"
        alt="EnginZyme Logo"
        src="../../assets/logo_dark.svg"
      />
      <img
        v-if="showImage && !dark"
        alt="EnginZyme Logo"
        src="../../assets/logo_light.svg"
      />
      <span class="app-banner__app-name">OPiDi</span>
      <span class="app-banner__app-version">{{ version }}</span>
    </div>
  </div>
</template>

<script>
import appData from "../../../package.json";
/**
 * A component that renders the app banner.
 */
export default {
  name: "AppBanner",
  props: {
    /**
     * The color theme of the banner.
     */
    dark: {
      type: Boolean,
      default: false,
    },
    /**
     * Determines if the log will be present in the render or not.
     */
    showImage: {
      type: Boolean,
      default: true,
    },
  },
  data() {
    return {
      /**
       * The version of the application.
       */
      version: appData.version,
    };
  },
};
</script>

<style scoped>
.app-banner__app-name {
  margin-left: 1em;
  font-style: normal;
  font-weight: 1000;
  font-size: 20px;
}
.app-banner__app-version {
  margin-left: 0.5em;
  font-size: 10px;
  font-weight: 700;
}
.app-banner--dark {
  display: flex;
  flex-direction: row;
  align-items: center;
  color: #070b1f;
  opacity: 1;
}
.app-banner--light {
  display: flex;
  flex-direction: row;
  align-items: center;
  color: #ffffff;
  opacity: 1;
}
.app-banner__grid {
  display: grid;
  align-items: center;
}
</style>
