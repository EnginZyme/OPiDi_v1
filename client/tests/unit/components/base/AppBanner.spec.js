import { shallowMount } from "@vue/test-utils";
import AppBanner from "@/components/base/AppBanner";

describe("UnshallowMounted AppBanner", () => {
  it("has data", () => {
    expect(typeof AppBanner.data).toBe("function");
  });
});

describe("shallowMounted AppBanner", () => {
  const wrapper = shallowMount(AppBanner);

  it("has an image", () => {
    expect(wrapper.find("img").exists()).toBe(true);
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe("shallowMounted AppBanner with 'showImage' prop set as false", () => {
  const wrapper = shallowMount(AppBanner, {
    propsData: {
      showImage: false,
    },
  });

  it("has no logo image", () => {
    expect(wrapper.find("img").exists()).toBe(false);
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe("shallowMounted AppBanner with 'dark' prop set as true", () => {
  const wrapper = shallowMount(AppBanner, {
    propsData: {
      dark: true,
    },
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe("shallowMounted AppBanner with 'version' property set to a random value", () => {
  const wrapper = shallowMount(AppBanner);

  const major = Math.floor(Math.random() * 100);
  const minor = Math.floor(Math.random() * 100);
  const patch = Math.floor(Math.random() * 100);
  const randomVersionValue = `${major}.${minor}.${patch}`;

  wrapper.setData({
    version: randomVersionValue,
  });

  it("has the random version value in the rendered text", () => {
    expect(wrapper.text()).toContain(randomVersionValue);
  });
});
