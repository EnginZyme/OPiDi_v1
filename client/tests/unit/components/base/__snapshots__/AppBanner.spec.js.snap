// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallowMounted AppBanner renders correctly 1`] = `
<div
  class="app-banner__grid"
>
  <div
    class="app-banner--light"
  >
    <!---->
     
    <img
      alt="EnginZyme Logo"
      src="../../assets/logo_light.svg"
    />
     
    <span
      class="app-banner__app-name"
    >
      Protocol Designer
    </span>
     
    <span
      class="app-banner__app-version"
    >
      2.1.1
    </span>
  </div>
</div>
`;

exports[`shallowMounted AppBanner with 'dark' prop set as true renders correctly 1`] = `
<div
  class="app-banner__grid"
>
  <div
    class="app-banner--dark"
  >
    <img
      alt="EnginZyme Logo"
      src="../../assets/logo_dark.svg"
    />
     
    <!---->
     
    <span
      class="app-banner__app-name"
    >
      Protocol Designer
    </span>
     
    <span
      class="app-banner__app-version"
    >
      2.1.1
    </span>
  </div>
</div>
`;

exports[`shallowMounted AppBanner with 'showImage' prop set as false renders correctly 1`] = `
<div
  class="app-banner__grid"
>
  <div
    class="app-banner--light"
  >
    <!---->
     
    <!---->
     
    <span
      class="app-banner__app-name"
    >
      Protocol Designer
    </span>
     
    <span
      class="app-banner__app-version"
    >
      2.1.1
    </span>
  </div>
</div>
`;
